client/state/posts/selectors.js

```diff
@@ -34,23 +34,32 @@ export function getPost( state, globalId ) {
 }
 
 /**
- * Returns a normalized post object by its global ID.
+ * Returns a normalized post object by its global ID, or null if the post does
+ * not exist. A normalized post includes common transformations to prepare the
+ * post for display.
  *
- * @param  {Object} state    Global state tree
- * @param  {String} globalId Post global ID
- * @return {Object}          Post object
+ * @param  {Object}  state    Global state tree
+ * @param  {String}  globalId Post global ID
+ * @return {?Object}          Post object
  */
 export const getNormalizedPost = createSelector(
 	( () => {
-		// Cache normalize flow in immediately-invoked closure so to avoid
+		// Cache normalize flow in immediately-invoked closure to avoid
 		// regenerating same flow on each call to this selector
 		const normalize = flow( [
 			firstPassCanonicalImage,
 			decodeEntities,
 			stripHtml
 		] );
 
-		return ( state, globalId ) => normalize( cloneDeep( getPost( state, globalId ) ) );
+		return ( state, globalId ) => {
+			const post = getPost( state, globalId );
+			if ( ! post ) {
+				return null;
+			}
+
+			return normalize( cloneDeep( post ) );
```

Comment:
and i agree, this is a nice thing for a selector, but isn't this going to throw away all of the normalized posts when any one changes?